phantom/env.py

```diff
@@ -74,6 +74,11 @@ def __init__(
         self._samplers: List[Sampler] = []
 
         if env_supertype is not None:
+            if isinstance(env_supertype, dict):
+                env_supertype = self.Supertype(**env_supertype)
```

Comment:
here as well, we assume that the Supertype inner class is present

Reply:
Yes. If the supertype values are passed as a dict then we need a `Supertype` inner class to check against and build the supertype instance. The idea of this is that when evaluating a policy/environment the agent class that the supertype belongs to may not be in scope. Instead we can pass the values and ranges in as a dict.